src/software/thunderscope/field/simulator_layer.py

```diff
@@ -0,0 +1,57 @@
+import software.python_bindings as geom
+import pyqtgraph as pg
+from proto.import_all_protos import *
+from pyqtgraph.Qt import QtCore, QtGui
+from pyqtgraph.Qt.QtCore import Qt
+
+from software.thunderscope.colors import Colors
+from software.networking.threaded_unix_listener import ThreadedUnixListener
+from software.thunderscope.constants import (
+    BALL_RADIUS,
+    MM_PER_M,
+)
+from software.thunderscope.field.field_layer import FieldLayer
+from software.thunderscope.field.world_layer import WorldLayer
+from software.thunderscope.thread_safe_buffer import ThreadSafeBuffer
+from extlibs.er_force_sim.src.protobuf.world_pb2 import (
+    SimulatorState,
+    SimBall,
+    SimRobot,
+)
+
+
+class SimulatorLayer(FieldLayer):
+    def __init__(self, friendly_colour_yellow, buffer_size=5):
```

Comment:
request: the ability to control all the buffer sizes from a command line arg

Reply:
added